source/validate_id.cpp

```diff
@@ -3044,7 +3044,7 @@ function<bool(unsigned)> getCanBeForwardDeclaredFunction(SpvOp opcode) {
       break;
 
     case SpvOpPhi:
-      out = [](unsigned index) { return index > 1; };
+      out = [](unsigned index) { return index > 1 && index % 2 == 1; };
```

Comment:
No so.
The point of an OpPhi is to bring value from a later block into an earlier block, when you can't yet "see" those later definitions.


  %n = OpPhi %int %next_n %label %zero %entry
   ...
   ; loop variable update.
   %next_n = OpIAdd %int %n %one  ; this is a forward declaration 
```